server/actions/serveFile.js

```diff
@@ -21,184 +21,214 @@ const AutoIndex = !process.env.DISABLE_INDEX;
+
+			res
+				.status(500)
+				.type("text")
+				.send(
+					`Cannot generate module for ${req.packageSpec}${
+						req.filename
+					}\n\n${debugInfo}`
+				);
+		} else {
+			// Cache modules for 1 year.
+			res
+				.set({
+					"Content-Type": "application/javascript; charset=utf-8",
+					"Content-Length": Buffer.byteLength(code),
+					"Cache-Control": "public, max-age=31536000",
+					"Cache-Tag": "file,js-file,js-module"
+				})
+				.send(code);
+		}
+	});
 }
 
 function serveStaticFile(req, res) {
-  const tags = ["file"];
-
-  const ext = path.extname(req.filename).substr(1);
-  if (ext) {
-    tags.push(`${ext}-file`);
-  }
-
-  let contentType = getFileContentType(req.filename);
-  if (contentType === "application/javascript") {
-    contentType += "; charset=utf-8";
-  }
-
-  // Cache files for 1 year.
-  res.set({
-    "Content-Type": contentType,
-    "Content-Length": req.stats.size,
-    "Cache-Control": "public, max-age=31536000",
-    "Last-Modified": req.stats.mtime.toUTCString(),
-    ETag: etag(req.stats),
-    "Cache-Tag": tags.join(",")
-  });
-
-  const file = path.join(req.packageDir, req.filename);
-  const stream = fs.createReadStream(file);
-
-  stream.on("error", error => {
-    console.error(`Cannot send file ${req.packageSpec}${req.filename}`);
-    console.error(error);
-    res.sendStatus(500);
-  });
-
-  stream.pipe(res);
+	const tags = ["file"];
+
+	const ext = path.extname(req.filename).substr(1);
+	if (ext) {
+		tags.push(`${ext}-file`);
+	}
+
+	let contentType = getFileContentType(req.filename);
+	if (contentType === "application/javascript") {
+		contentType += "; charset=utf-8";
+	}
+
+	// Cache files for 1 year.
+	res.set({
+		"Content-Length": req.stats.size,
+		"Cache-Control": "public, max-age=31536000",
+		"Last-Modified": req.stats.mtime.toUTCString(),
+		ETag: etag(req.stats),
+		"Cache-Tag": tags.join(",")
+	});
+
+	const file = path.join(req.packageDir, req.filename);
+	const stream = fs.createReadStream(file);
+	
+	stream.on("error", error => {
+		console.error(`Cannot send file ${req.packageSpec}${req.filename}`);
+		console.error(error);
+		res.sendStatus(500);
+	});
+
+	let chunks = [];
+	let codeString = '';
+	// push file content chunk-wise from stream to array
+	stream.on("data", chunk => {
+		chunks.push(chunk);
+	});
+
+	// return syntax highlighted code as html 
+	stream.on("close", () => {
+
+		// turn chunk-array to string
+		codeString = Buffer.concat(chunks).toString();
+		// html template using highlight.js
+		let highlightStyle = "default";
+		let html = ` 
+			<html>
+			<head>
+				<link rel="stylesheet" type="text/css" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/${highlightStyle}.min.css">
+				<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
+				<script src="//cdnjs.cloudflare.com/ajax/libs/highlightjs-line-numbers.js/2.3.0/highlightjs-line-numbers.min.js"></script>
```

Comment:
you probably want to use unpkg for that ;)

Reply:
Yeah, thanks haha :+1: